builtInFunctions/esdtNFTTransfer.go

```diff
@@ -343,23 +343,25 @@ func (e *esdtNFTTransfer) processNFTTransferOnSenderShard(
 	return vmOutput, nil
 }
 
-func hasDynamicRole(account vmcommon.UserAccountHandler, tokenID []byte, marshaller vmcommon.Marshalizer) (bool, error) {
+func hasDynamicRole(account vmcommon.UserAccountHandler, tokenID []byte, marshaller vmcommon.Marshalizer, enableEpochsHandler vmcommon.EnableEpochsHandler) (bool, error) {
 	roleKey := append(roleKeyPrefix, tokenID...)
 	roles, _, err := getESDTRolesForAcnt(marshaller, account, roleKey)
 	if err != nil {
 		return false, err
 	}
 
 	dynamicRoles := [][]byte{
-		[]byte(core.ESDTRoleNFTAddURI),
-		[]byte(core.ESDTRoleNFTUpdateAttributes),
 		[]byte(core.ESDTMetaDataRecreate),
 		[]byte(core.ESDTRoleNFTUpdate),
 		[]byte(core.ESDTRoleModifyCreator),
 		[]byte(core.ESDTRoleModifyRoyalties),
 		[]byte(core.ESDTRoleSetNewURI),
 	}
 
+	if enableEpochsHandler.IsFlagEnabled(DynamicEsdtFlag) {
+		dynamicRoles = append(dynamicRoles, []byte(core.ESDTRoleNFTAddURI), []byte(core.ESDTRoleNFTUpdateAttributes))
```

Comment:
there are no unit tests for checking the dynamicRoles
core.ESDTRoleNFTAddURI,
core.ESDTRoleNFTUpdateAttributes

Reply:
Added.